run.go

```diff
@@ -22,25 +26,49 @@ func Run() error {
 		service.Registry.Register(service)
 	}
 
+	// Register probes for internal services
+	grpcStarted := RegisterHealthProbe("grpc", func(ctx context.Context) error {
+		return nil
+	}, func(ctx context.Context) error {
+		if !service.grpcRunning {
+			return errors.New("not running")
+		}
+		return nil
+	})
+	prometheusStarted := RegisterHealthProbe("prometheus", func(ctx context.Context) error {
+		return nil
+	}, func(ctx context.Context) error {
+		if !service.prometheusRunning {
```

Comment:
I wonder whether a failing prometheus server  should causes the reboot of our entire pod and thus core application?

Reply:
🤷 Yeah was thinking the same, I arrived at it being unlikely that both pods would have prometheus fail at once so it would be better to get the other capturing data again instead of just leaving it. Maybe that's not correct though as it's not a fundamental part of the application?